ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,529 @@
+		monitortype = Mtype.(string)
+	} else {
+		monitortype = "HTTP"
+	}
+
+	if monitortype == "HTTP" {
+		if Mport, ok := d.GetOk(pdnsGlbMonitorPort); ok {
+			monitorport = int64(Mport.(int))
+		} else {
+			monitorport = 80
+		}
+	} else if monitortype == "HTTPS" {
+		if Mport, ok := d.GetOk(pdnsGlbMonitorPort); ok {
+			monitorport = int64(Mport.(int))
+		} else {
+			monitorport = 443
+		}
+	} else if monitortype == "TCP" {
+		if Mport, ok := d.GetOk(pdnsGlbMonitorPort); ok {
+			monitorport = int64(Mport.(int))
+		} else {
+			return fmt.Errorf("Error Monitor port should be provided for TCP")
+		}
+	}
+	CreateMonitorOptions.SetName(monitorname)
+	CreateMonitorOptions.SetType(monitortype)
+	CreateMonitorOptions.SetPort(monitorport)
+	CreateMonitorOptions.SetInterval(monitorinterval)
+	CreateMonitorOptions.SetRetries(monitorretries)
+	CreateMonitorOptions.SetTimeout(monitortimeout)
+
+	//Setting HTTP,HTTPS,TCP specific parameters
+	monitorpath, pathok := d.GetOk(pdnsGlbMonitorPath)
+	monitorexpectedcodes, expectedcodeok := d.GetOk(pdnsGlbMonitorExpectedCodes)
+	monitorallowinsecure, allowinsecureok := d.GetOk(pdnsGlbMonitorAllowInsecure)
+	monitormethod, methodok := d.GetOk(pdnsGlbMonitorMethod)
+	monitorexpectedbody, expectedbodyok := d.GetOk(pdnsGlbMonitorExpectedBody)
+	monitorheaders, headerok := d.GetOk(pdnsGlbMonitorHeaders)
+
+	if (monitortype == "HTTP") || (monitortype == "HTTPS") {
+		if pathok {
+			CreateMonitorOptions.SetPath((monitorpath).(string))
+		} else {
+			CreateMonitorOptions.SetPath("/")
+		}
+		if expectedcodeok {
+			CreateMonitorOptions.SetExpectedCodes((monitorexpectedcodes).(string))
+		} else {
+			CreateMonitorOptions.SetExpectedCodes("200")
+		}
+		if methodok {
+			CreateMonitorOptions.SetMethod((monitormethod).(string))
+		} else {
+			CreateMonitorOptions.SetMethod("GET")
+		}
+		if expectedbodyok {
+			CreateMonitorOptions.SetExpectedBody((monitorexpectedbody).(string))
+		}
+		if allowinsecureok {
+			if monitortype == "HTTPS" {
+				CreateMonitorOptions.SetAllowInsecure((monitorallowinsecure).(bool))
+			} else {
+				return fmt.Errorf("Monitor allow_insecure is not supported in type HTTP")
+			}
+		} else {
+			CreateMonitorOptions.SetAllowInsecure(false)
+		}
+		if headerok {
+			expandedmonitorheaders, err := expandGLBMonitors(monitorheaders, pdnsGlbMonitorHeadersName)
+			if err != nil {
+				return err
+			}
+			CreateMonitorOptions.SetHeadersVar(expandedmonitorheaders)
+		}
+
+	} else {
+		if pathok || expectedcodeok || allowinsecureok || methodok || expectedbodyok || headerok {
+			return fmt.Errorf("Monitor path/expected_codes/expected_body/allow_insecure/method/headers is not supported in type TCP")
+		}
+	}
+	response, detail, err := sess.CreateMonitor(CreateMonitorOptions)
+	if err != nil {
+		return fmt.Errorf("Error creating pdns GLB monitor:%s\n%s", err, detail)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", instanceID, *response.ID))
+	d.Set(pdnsGlbMonitorID, *response.ID)
```

Comment:
add this line into Read(). remove it from here.

Reply:
handled